compute_tools/src/compute.rs

```diff
@@ -2123,6 +2216,158 @@ LIMIT 100",
             info!("Pageserver config changed");
         }
     }
+
+    // If prewarm failed, we want to get overall number of segments as well as done ones.
+    // However, this function should be reliable even if querying postgres failed.
+    pub async fn prewarm_status(&self) -> PrewarmStatus {
```

Comment:
NIT: this `compute.rs` is really huge already. Should we move all new method implementations and structs into `compute_prewarm.rs`? It will be pretty well-scoped and should improve navigation and readability. WDYT?

Personally, I was already struggling with the `compute.rs` size

Reply:
It seems like we have a good pattern going where the `ComputeNode` methods become small wrappers around functions in other files. I agree with this suggestion.

Reply:
Oh, I can't remember if it is, but that is also a good idea!

Reply:
Fixed